src/libraries/System.Formats.Asn1/tests/Writer/Asn1WriterTests.cs

```diff
@@ -25,13 +25,17 @@ internal static void Verify(AsnWriter writer, string expectedHex)
             Span<byte> dest = encoded2.AsSpan(0, encoded.Length - 1);
             Assert.False(writer.TryEncode(dest, out int bytesWritten), "writer.TryEncode (too small)");
             Assert.Equal(0, bytesWritten);
+            AssertExtensions.Throws<ArgumentException>("destination", () => writer.Encode(encoded2.AsSpan(0, encoded.Length - 1)));
             Assert.Equal(255, encoded2[0]);
             Assert.Equal(254, encoded2[encoded.Length]);
 
             dest = encoded2.AsSpan(0, encoded.Length);
             Assert.True(writer.TryEncode(dest, out bytesWritten), "writer.TryEncode (exact length)");
             Assert.Equal(encoded.Length, bytesWritten);
-            Assert.True(dest.SequenceEqual(encoded), "dest.SequenceEqual(encoded2) (exact length)");
+            Assert.True(dest.SequenceEqual(encoded), "dest.SequenceEqual(encoded2) (exact length) from TryEncode");
```

Comment:
Perhaps using `Assert.Equal(dest.ToArray(), encoded.ToArray())` might be better in identifying errors?

Reply:
Assuming that the array mismatch pretty-printer has been fixed to show where the error was (it used to always show the first N elements, even if the error was beyond N), yeah -- or ByteArrayToHex.  In this case, `encoded` was already compared using a nice-to-diagnose flow, and the other variants are using one that's a little nicer on the GC since failure isn't expected -- once that burns me it'll change.  (This complex routine is run for every single writer output-producing test case)